dev/sql/schema_relational.sql

```diff
@@ -48,26 +59,61 @@ begin
 end;
 $not_null_trigger$ language plpgsql;
 
-CREATE FUNCTION notify_modified_models() RETURNS trigger AS $notify_trigger$
+CREATE FUNCTION log_modified_models() RETURNS trigger AS $log_notify_trigger$
 DECLARE
-    channel TEXT;
+    operation TEXT;
     payload TEXT;
 BEGIN
-    channel:= LOWER(TG_OP);
+    operation := LOWER(TG_OP);
     payload := TG_TABLE_NAME || '/' || NEW.id;
     IF (TG_OP = 'DELETE') THEN
         payload = TG_TABLE_NAME || '/' || OLD.id;
     END IF;
 
-    PERFORM pg_notify(channel, payload);
-    INSERT INTO os_notify_log_t (channel, payload, xact_id, timestamp) VALUES (channel, payload, pg_current_xact_id(), 'now');
+    INSERT INTO os_notify_log_t (operation, payload, xact_id, timestamp) VALUES (operation, payload, pg_current_xact_id(), 'now');
+    RETURN NULL;  -- AFTER TRIGGER needs no return
+END;
+$log_notify_trigger$ LANGUAGE plpgsql;
+
+CREATE FUNCTION notify_modified_models() RETURNS trigger AS $notify_trigger$
+DECLARE
+    pl TEXT;
+    body_content_text TEXT;
+BEGIN
+    -- Running the trigger for the first time in a transaction the table is created and dropped after commiting the transaction.
+    -- Every next run of the trigger in this transaction raises a notice that teh table exists. This is not ideal.
+    CREATE LOCAL TEMPORARY UNLOGGED TABLE
+    IF NOT EXISTS tbl_notify_counter_tx_once (
+        "id" integer NOT NULL PRIMARY KEY GENERATED ALWAYS AS IDENTITY
+    ) ON COMMIT DROP;
+
+    -- If running for the first time, select all modification notifications for the current transaction and send them to os-notify.
+    IF NOT EXISTS (SELECT * FROM tbl_notify_counter_tx_once) THEN
+        INSERT INTO tbl_notify_counter_tx_once DEFAULT VALUES;
+        -- Get all modifications as fqid of the current transaction and format them using a comma separated list.
+        SELECT array_to_string(
+            ARRAY (
+                SELECT payload
+                    FROM os_notify_log_t
+                    WHERE xact_id = pg_current_xact_id()),
+                '","')
+            INTO body_content_text;
+        -- Concat current transaction id and formated fqid list to a json object.
+        pl := '{"xactId":' ||
+            pg_current_xact_id() ||
+            ',"fqids":["' ||
+            body_content_text ||
+            '"]}';
+        PERFORM pg_notify('os-notify', pl);
```

Comment:
I am not able to listen on 'os-notify'. It seems that `-` is not an allowed character. Could you rename this to `os_notify` or some other name?

The documentation says, the channel has to be `any identifier`. And "the internet" says, and postgres identifier can only be letters, numbers and underscore.

Reply:
Done. postgres itself seems to support `-`. Used psql to test this PR and had no problems with it.